src/edu/csus/ecs/pc2/core/InternalController.java

```diff
@@ -2659,7 +2661,80 @@ public void sendToTeams(Packet packet) {
             sendPacketToClients(packet, ClientType.Type.TEAM);
         }
     }
+    
+    public void sendToGroupsandIndividualTeams(Packet packet,IElementObject[] ultimateDestination) {
+
+//        Properties properties = (Properties) packet.getContent();
+//        // does the packet includes problemDataFiles
+//        boolean abort = true;
+//        if (properties.containsKey(PacketFactory.PROBLEM_DATA_FILES)) {
+//            // clone before start mucking with it, or do we need a deep clone?
+//            Properties cloneProperties = new Properties();
+//            for (Iterator<?> iter = properties.keySet().iterator(); iter.hasNext();) {
+//                String element = (String) iter.next();
+//                // skip PROBLEM_DATA_FILES, otherwise clone the element
+//                if (!element.equals(PacketFactory.PROBLEM_DATA_FILES)) {
+//                    cloneProperties.put(element, properties.get(element));
+//                    abort = false;
+//                }
+//            }
+//            packet = PacketFactory.clonePacket(packet.getSourceId(), packet.getDestinationId(), packet);
+//            // stick it back into the packet
+//            packet.setContent(cloneProperties);
+//        } else {
+//            abort = false;
+//        }
+//        if (!abort) {
+//            sendPacketToClients(packet, ClientType.Type.TEAM);
+//        }
+        
+        // sendToClient(ConnectionHandlerID connectionHandlerID, Packet packet
+        ClientId[] clientIds = contest.getLocalLoggedInClients(ClientType.Type.TEAM);
+        
+        List<ClientId> badClients = new ArrayList<ClientId>();
```

Comment:
Is badClients used anywhere?

Reply:
My understanding is that ClientId's where packet sending has failed must be collected. So I have to use that arraylist somehow.

Reply:
Rest is added